src/workerd/io/worker-entrypoint.c++

```diff
@@ -495,13 +502,13 @@ kj::Promise<WorkerInterface::ScheduledResult> WorkerEntrypoint::runScheduled(
 
   // Scheduled handlers run entirely in waitUntil() tasks.
   context.addWaitUntil(
-      context.run([scheduledTime, cron, entrypointName = entrypointName, &context,
+      context.run([this, scheduledTime, cron, entrypointName = entrypointName, &context,
                       &metrics = incomingRequest->getMetrics()](Worker::Lock& lock) mutable {
     TRACE_EVENT("workerd", "WorkerEntrypoint::runScheduled() run");
     jsg::AsyncContextFrame::StorageScope traceScope = context.makeAsyncTraceScope(lock);
 
-    lock.getGlobalScope().startScheduled(
-        scheduledTime, cron, lock, lock.getExportedHandler(entrypointName, context.getActor()));
+    lock.getGlobalScope().startScheduled(scheduledTime, cron, lock,
+        lock.getExportedHandler(entrypointName, kj::mv(props), context.getActor()));
```

Comment:
I guess this must be guaranteed somehow since we also capture `entrypointName` by ptr but what ensures that `this` (and thus `props`) has not been destroyed by the time this code executes?

Reply:
I think that for `WorkerEntrypoint` to be destroyed, first the promise returned by `runScheduled()` would have to have been canceled, which from what I could tell will actually drop the IoContext, canceling all waitUntil()s.

This is actually sort of inconsistent: Dropping the promise returned by request() (for HTTP) does NOT cancel waitUntil()s. But in that case, the code that uses `props` is not in a `waitUntil`, it's in a promise that _is_ still canceled when the request is canceled...

With all that said, I don't feel super-great about depending on this so I'm going to remove the `this` capture and instead capture `props` by-move.

Incidentally, the treatment of `entrypointName` doesn't seem to prove anything as it appears that `WorkerEntrypoint` as a whole does not own `entrypointName` but rather expects some higher power to keep it alive...